### DISCUSSION TITLE
Trackermount get_orientation

### DISCUSSION THREAD

**Discussion by jules-ch** [pvlib/pvlib-python#1492] [Q&A] [answered]
Hello,
I am using `pvsystem.AbstractMount` overiding `get_orientation` function.

I just have a quick question.
After investigating, it seems this function only needs to return `surface_azimuth` & `surface_tilt`. I know that `singleaxis.tracking` returns those values & keep `surface_tilt` positive & change `surface_azimuth` accordingly.

Is it mandatory to have `surface_tilt` postive to compute the model. I'd rather keep `surface_tilt` & `surface_azimuth` coherent with the right hand rule ?

**Comment by kandersolar** [comment, marked as answer]
Negative tilts might accidentally work in some pvlib models, but I don't think we make any promises about whether it works today and whether it will continue to work in the future.  So perhaps not absolutely mandatory, but you use negative tilts at your own risk.  In contrast to something like the `tracker_theta` from `pvlib.tracking.singleaxis` where the sign follows the RHR, I think of `surface_tilt` and `surface_azimuth` as being generic quantities that could have come from any mount.  #1480 might be of interest.

It doesn't matter for your question, but I'm curious what kind of custom mount are you modeling.  I wasn't sure `AbstractMount` would get much use in the community and am pleased to hear that at least one person is making use of it :)

**Reply by jules-ch** [reply]
Alright, I'm using custom mount to compute energy production with custom angles used for agrivoltaic systems.
Energy production is not the only focus and we can have tracking, partial tracking or panels that just anti-track the sun to provide maximum irradiance for crops under panels depending on weather conditions / plant growth / etc.

Thanks for the heads up, I'll try to keep surface tilt / azimuth the same as `singleaxis.tracking`.
I'll look to the PR you linked.

**Reply by jules-ch** [reply]
I'll wait for 0.9.2 release & we can use the new function `calc_surface_orientation`.
Thanks for the quick answer @kanderso-nrel

**Reply by kandersolar** [reply]
No problem and thanks for the info!  FYI I think the plan is to release 0.9.2 in perhaps a month or so, although we haven't set a firm date yet.
